packages/nextjs/src/app-beta/ClerkProvider.tsx

```diff
@@ -1,9 +1,18 @@
+import { ClerkProviderProps } from '@clerk/clerk-react';
 import React from 'react';
 
 import { initialState } from './auth';
 import { ClerkProvider as ClerkProviderClient } from './client/ClerkProvider';
 
-export function ClerkProvider({ children }: { children: React.ReactNode }) {
+export function ClerkProvider({ children, ...restProps }: ClerkProviderProps) {
   const state = initialState()?.__clerk_ssr_state || { sessionId: null, orgId: null, userId: null };
-  return <ClerkProviderClient initialState={state}>{children}</ClerkProviderClient>;
+
+  return (
+    <ClerkProviderClient
+      {...restProps}
```

Comment:
Many thanks for the contribution! :)

This all looks good to me, but I'm not sure how the `appearance` prop will behave on a server-side component like this one. This prop accepts a value that might not be serializable (`appearance.elements` can also be a function). I'll run some tests to verify and get back to you ASAP

Reply:
That's make sense. Sorry to not having that investigated before, in that case maybe the component should still reuse `ClerkProviderProps` but "proxied" with `Omit<ClerkProviderProps,"not-supported-props">`?

Reply:
After this (maybe stale) discussion I've decided to narrow down the scope of this pull request.

This will allow to enable localization on server side components by reusing the same `ClerkProviderProps` but with a reduced api surface with `Pick<T,...>` - that way we can enable what we are sure to be able to serialize.

Does this looks good to you?

Obv: I will push some other commits and squash locally to fix pipeline/test issues.